sycl/include/sycl/ext/oneapi/experimental/enqueue_functions.hpp

```diff
@@ -0,0 +1,327 @@
+//==------ enqueue_functions.hpp ------- SYCL enqueue free functions -------==//
+//
+// Part of the LLVM Project, under the Apache License v2.0 with LLVM Exceptions.
+// See https://llvm.org/LICENSE.txt for license information.
+// SPDX-License-Identifier: Apache-2.0 WITH LLVM-exception
+//
+//===----------------------------------------------------------------------===//
+
+#pragma once
+
+#include <utility> // for std::forward
+
+#include <sycl/event.hpp>                            // for event
+#include <sycl/ext/oneapi/properties/properties.hpp> // for properties
+#include <sycl/handler.hpp>                          // for handler
+#include <sycl/nd_range.hpp>                         // for nd_range
+#include <sycl/queue.hpp>                            // for queue
+#include <sycl/range.hpp>                            // for range
```

Comment:
is that the new guideline? 
I've never liked the "for " part as it is redundant. But I find the mention of why a header is being included useful.

Reply:
Right, I forgot. I am fine with that as they are generally a little awkward to track. I will leave the discussion open in case you would like to continue.